compatibility/src/lib.rs

```diff
@@ -0,0 +1,98 @@
+#![cfg(test)]
+
+use ::rand::Rng;
+use bincode_1::Options;
+
+mod rand;
+mod sway;
+
+pub fn test_same_with_config<T, C, O>(t: &T, bincode_1_options: O, bincode_2_config: C)
+where
+    T: bincode_2::Encode
+        + bincode_2::Decode
+        + serde::Serialize
+        + serde::de::DeserializeOwned
+        + core::fmt::Debug
+        + PartialEq,
+    C: bincode_2::config::Config,
+    O: bincode_1::Options + Copy,
+{
+    let bincode_1_output = bincode_1_options.serialize(t).unwrap();
+    let bincode_2_output = bincode_2::encode_to_vec(t, bincode_2_config).unwrap();
+
+    assert_eq!(
+        bincode_1_output, bincode_2_output,
+        "{:?} serializes differently",
+        t
+    );
+
+    let decoded: T = bincode_1_options.deserialize(&bincode_1_output).unwrap();
+    assert_eq!(&decoded, t);
+    let decoded: T = bincode_1_options.deserialize(&bincode_2_output).unwrap();
+    assert_eq!(&decoded, t);
+
+    let decoded: T = bincode_2::decode_from_slice(&bincode_1_output, bincode_2_config)
+        .unwrap()
+        .0;
+    assert_eq!(&decoded, t);
+    let decoded: T = bincode_2::decode_from_slice(&bincode_2_output, bincode_2_config)
+        .unwrap()
+        .0;
+    assert_eq!(&decoded, t);
+}
+
+pub fn test_same<T>(t: T)
+where
+    T: bincode_2::Encode
+        + bincode_2::Decode
+        + serde::Serialize
+        + serde::de::DeserializeOwned
+        + core::fmt::Debug
+        + PartialEq,
+{
+    test_same_with_config(
```

Comment:
We should add a test that verifies that bare `legacy` works with the default options for bincode 1

Reply:
It does not. I will add a test that compares `bincode2::config::legacy()` with `bincode_1::DefaultOptions::new().with_fixint_encoding()` as this is what bincode 1 uses internally